app/redux/selectors/SetsSelectors.js

```diff
@@ -259,4 +261,50 @@ export const hasChangesToSync = (state) => {
     return (root.setIDsToUpload.length > 0 || root.setIDsBeingUploaded.length > 0);
 };
 
+// collapsed metrics
+
+export const getBestAvgVelocityEver = (state, set) => {
+    let historySets = getHistorySetsChronological(state);
+    let workoutSets = getWorkoutSets(state);
+
+    // find all instances of this exercise with weight and reps
+    let matchedSets = historySets.concat(workoutSets).filter(historySet => areSetsComparable(historySet, set));
+    
+    let avgVs = matchedSets.map((matchedSet) => {
+        return CollapsedMetrics.getAvgVelocities(matchedSet);
+    });
+
+    avgVs = avgVs.reduce((a, b) => a.concat(b), []);
+    
+    if (avgVs.length > 0) {
+        return Math.max(...avgVs);
+    } else {
+        return null;
+    }
+};
+
+export const getBestROMEver = (state, set) => {
```

Comment:
- [ ] Do a get best ever and pass in function. So make a generic get best ever that reqs a comparison functino, and then the actual individual functions pass in the generic function